src/main/java/bisq/core/dao/blockchain/BsqBlockChain.java

```diff
@@ -599,25 +620,50 @@ public long getProposalFee(int blockHeight) {
         });
     }
 
-    //TODO not impl yet
     @Override
-    public boolean isProposalPeriodValid(int blockHeight) {
-        return lock.read(() -> true);
+    public long getBlindVoteFee(int blockHeight) {
+        return lock.read(() -> getValueAtHeight(blindVoteFees, blockHeight));
     }
 
     @Override
-    public long getBlindVoteFee(int blockHeight) {
+    public long getQuorum(int blockHeight) {
+        return lock.read(() -> getValueAtHeight(quorumByHeight, blockHeight));
+    }
+
+    @Override
+    public long getVoteThreshold(int blockHeight) {
+        return lock.read(() -> getValueAtHeight(voteThresholdByHeight, blockHeight));
+    }
+
+    @Override
+    public long getVoteThreshold() {
+        return lock.read(() -> getVoteThreshold(chainHeadHeight));
+    }
+
+    @Override
+    public long getQuorum() {
+        return lock.read(() -> getQuorum(chainHeadHeight));
+    }
+
+    private long getValueAtHeight(Set<Tuple2<Long, Integer>> set, int blockHeight) {
         return lock.read(() -> {
-            long fee = -1;
-            for (Tuple2<Long, Integer> feeAtHeight : blindVoteFees) {
-                if (feeAtHeight.second <= blockHeight)
-                    fee = feeAtHeight.first;
+            long value = -1;
+            for (Tuple2<Long, Integer> currentValue : set) {
+                if (currentValue.second <= blockHeight)
+                    value = currentValue.first;
```

Comment:
Iterating over a set doesn't guarantee order so this would not work, unless do something like
```
Tuple2<Long, Integer> value = new Tuple2<>(0,0);
for (Tuple2<Long, Integer> currentValue : set) {
  if (currentValue.second <= blockHeight && value.second < currentValue.second)
    value = currentValue;
}
```